nodejs/dist/uws.js

```diff
@@ -1,98 +1,134 @@
-    var self = this;
-    this.nativeSocket = nativeSocket;
-    this.server = server;
-
-    this.send = function(message, options) {
-        if (self.nativeSocket !== null) {
-            var binary = false;
-            if (options !== undefined) {
-                binary = options.binary;
-            }
-            self.server.nativeServer.send(self.nativeSocket, message, binary);
-        } else {
-            /* ignore sends on closed sockets */
-        }
+
+const uws = require(`./uws_${process.platform}_${process.versions.modules}`);
+const NativeServer = uws.Server;
+
+class Socket extends EventEmitter {
+    /**
+     * Creates a Socket instance.
+     *
+     * @param {Object} nativeSocket Socket instance
+     * @param {Server} server Server instance
+     */
+    constructor(nativeSocket, server) {
+        super();
+        this.nativeSocket = nativeSocket;
+        this.server = server;
+    }
+
+    /**
+     * Sends a message.
+     *
+     * @param {String|Buffer} message The message to send
+     * @param {Object} options Send options
+     * @return {Boolean} `true` if the message has been sent, else `false`
+     * @public
+     */
+    send(message, options) {
+        /* ignore sends on closed sockets */
+        if (!this.nativeSocket) return false;
+
+        const binary = options && options.binary || false;
+        this.server.nativeServer.send(this.nativeSocket, message, binary);
+        return true;
     }
 
-    this.close = function() {
-        if (self.nativeSocket !== null) {
-            self.server.nativeServer.close(self.nativeSocket);
-            self.nativeSocket = null;
-        } else {
-            /* ignore close on closed sockets */
-        }
+    /**
+     * Closes the socket.
+     *
+     * @public
+     */
+    close() {
+        /* ignore close on closed sockets */
+        if (!this.nativeSocket) return;
+
+        this.server.nativeServer.close(this.nativeSocket);
+        this.server = this.nativeSocket = undefined;
     }
 }
 
-module.exports.Server = function Server(options) {
-    EventEmitter.call(this);
+class Server extends EventEmitter {
+    /**
+     * Creates a Server instance.
+     *
+     * @param {Object} options Configuration options.
+     */
+    constructor(options) {
+        super();
+        /* we currently do only support port as option */
+        this.nativeServer = new NativeServer(options.port);
 
-    /* we currently do only support port as option */
-    this.nativeServer = new NativeServer(options.port);
-    var self = this;
+        this.nativeServer.onConnection((nativeSocket) => {
+            const socket = new Socket(nativeSocket, this);
+            this.nativeServer.setData(nativeSocket, socket);
+            this.emit('connection', socket);
+        });
 
-    this.close = function () {
-        self.nativeServer.close();
+        this.nativeServer.onDisconnection((nativeSocket) => {
+            const socket = self.nativeServer.getData(nativeSocket);
```

Comment:
Yes correct, I mostly only copy pasted this part, I missed it :)
I'm gonna fix it.